test_common/harness/testHarness.c

```diff
@@ -797,8 +797,43 @@ void checkDeviceTypeOverride( cl_device_type *inOutType )
             *inOutType = CL_DEVICE_TYPE_ACCELERATOR;
         else if( strcmp( force_cpu, "CL_DEVICE_TYPE_DEFAULT" ) == 0 )
             *inOutType = CL_DEVICE_TYPE_DEFAULT;
-    }
-
+    } else {
+        //Detect the device type
+		cl_platform_id  platform;
+		int error = clGetPlatformIDs(1, &platform, NULL);
```

Comment:
I think this should check for `CL_PLATFORM_INDEX` vs. unconditionally choosing the first platform?